compiler/noirc_evaluator/src/ssa/opt/constant_folding/result_cache.rs

```diff
@@ -191,6 +213,26 @@ impl ResultCache {
 
 #[derive(Debug)]
 pub(super) enum CacheResult<'a> {
-    Cached(&'a [ValueId]),
-    NeedToHoistToCommonBlock(BasicBlockId),
+    /// The result of an earlier instruction can be readily reused, because it was found
+    /// in a block that dominates the one where the current instruction is. We can drop
+    /// the current instruction and redefine its results in terms of the existing values.
+    Cached {
+        /// The block which contains the instruction we can reuse.
+        #[allow(unused)]
+        origin: BasicBlockId,
```

Comment:
Why have this field if it is unused

Reply:
I had to add this, then remove it, then add it back for debugging. Once I used it in the other mechanism, but then it just serve debug purposes. Sure, I can remove it now that it's unused.